Formula/p/postgis-postgresql17.rb

```diff
@@ -0,0 +1,162 @@
+class PostgisPostgresql17 < Formula
```

Comment:
Still not sure on naming:
* `postgis-postgresql17`
* `postgis-pg17`
* `pg17-postgis` (MacPorts-style[^1], but usually not Homebrew style)

[^1]: https://ports.macports.org/port/pg16-postgis3/